lisp/ess-r-d.el

```diff
@@ -190,16 +190,57 @@
 (easy-menu-add-item inferior-ess-mode-menu nil ess-tracebug-menu "end-dev")
 
 
-;; modify S Syntax table:
-(setq R-syntax-table S-syntax-table)
+;; Inherit from the S syntax table:
+(setq ess-r-syntax-table (copy-syntax-table S-syntax-table))
+
+;; Letting Emacs treat backquoted names and %ops% as strings solves
+;; many problems with regard to nested strings and quotes
+(modify-syntax-entry ?` "\"" ess-r-syntax-table)
+(modify-syntax-entry ?% "\"" ess-r-syntax-table)
+
+;; Underscore is valid in R symbols
+(modify-syntax-entry ?_ "_" ess-r-syntax-table)
+
+(modify-syntax-entry ?: "." ess-r-syntax-table)
+(modify-syntax-entry ?@ "." ess-r-syntax-table)
+(modify-syntax-entry ?$ "." ess-r-syntax-table)
+
+;; Prevent string delimiting characters from messing up output in the
+;; inferior buffer
+(setq inferior-ess-r-syntax-table (copy-syntax-table ess-r-syntax-table))
+(modify-syntax-entry ?\' "." inferior-ess-r-syntax-table)
+(modify-syntax-entry ?\" "." inferior-ess-r-syntax-table)
+(modify-syntax-entry ?` "." inferior-ess-r-syntax-table)
+(modify-syntax-entry ?% "." inferior-ess-r-syntax-table)
+
+(defun ess-r-font-lock-syntactic-face-function (state)
```

Comment:
> I am extracting indentation and font-lock pieces into a stand alone r-mode,

Great!

I don't remember exactly as I wrote this two months ago. I think this is a corollary of treating %op% and backquoted names as syntactic strings. In general I don't like dealing with paired identifiers with regexes and I think this function is more maintainable and extensible. We now have a more accurate and more general fontification of function calls and defuns (see screenshot attached in the OP).